packages/grafana-data/src/types/panel.ts

```diff
@@ -112,6 +117,35 @@ export interface PanelProps<T = any> {
   onChangeTimeRange: (timeRange: AbsoluteTimeRange) => void;
 }
 
+export interface ExplorePanelProps {
+  graphStyle: ExploreGraphStyle;
+  onChangeGraphStyle: (style: ExploreGraphStyle) => void;
+  data: DataFrame[];
+  absoluteRange: AbsoluteTimeRange;
+  range: TimeRange;
+  timeZone: TimeZone;
+  splitOpen: SplitOpen;
+  annotations?: DataFrame[];
+  loadingState: LoadingState;
+  loading?: boolean;
+  theme: GrafanaTheme2;
+  onUpdateTimeRange: (timeRange: AbsoluteTimeRange) => void;
+  width: number;
+  // TODO cannot import FilterItem from grafana/ui
+  // onCellFilterAdded: (filter: FilterItem) => void;
+  onCellFilterAdded: (filter: any) => void;
+  exploreId: ExploreId;
+  syncedTimes: boolean;
+  onClickFilterLabel: (key: string, value: string) => void;
+  onClickFilterOutLabel: (key: string, value: string) => void;
+  onStartScanning: () => void;
+  onStopScanning: () => void;
+  datasourceInstance?: DataSourceApi | null;
```

Comment:
no need to pass `| null` if the prop is optional.

Reply:
`datasourceInstance?: DataSourceApi` is equivalent to `datasourceInstance: DataSourceApi | undefined` but `null` isn't same as `undefined` so it needs be there from typing perspective as we pass `null` somewhere. At the same time we probably should not mix them this was just to quickly make build pass.